src/napari_spatialdata/_scatterwidgets.py

```diff
@@ -23,6 +29,81 @@
 ]
 
 
+class SelectFromCollection:
+    """
+    Select indices from a matplotlib collection using `LassoSelector`.
+
+    Selected indices are saved in the `ind` attribute. This tool fades out the
+    points that are not part of the selection (i.e., reduces their alpha
+    values). If your collection has alpha < 1, this tool will permanently
+    alter the alpha values.
+
+    Note that this tool selects collection objects based on their *origins*
+    (i.e., `offsets`).
+
+    Parameters
+    ----------
+    ax : `~matplotlib.axes.Axes`
+        Axes to interact with.
+    collection : `matplotlib.collections.Collection` subclass
+        Collection you want to select from.
+    alpha_other : 0 <= float <= 1
+        To highlight a selection, this tool sets all selected points to an
+        alpha value of 1 and non-selected points to *alpha_other*.
+    """
+
+    def __init__(
+        self,
+        viewer: Viewer,
+        model: ImageModel,
+        ax: Axes,
+        collection: Collection,
+        data: List[NDArrayA],
+        alpha_other: float = 0.3,
+    ):
+        self.viewer = viewer
+        self.model = model
+        self.canvas = ax.figure.canvas
+        self.collection = collection
+        self.alpha_other = alpha_other
+        self.exported_data = None
+        self.data = data
+        self.axes = ax
+
+        self.xys = collection.get_offsets()
+        self.Npts = len(self.xys)
+
+        # Ensure that we have separate colors for each object
+        self.fc = collection.get_facecolors()
+
+        if len(self.fc) == 0:
+            raise ValueError("Collection must have a facecolor")
+        elif len(self.fc) == 1:
+            self.fc = np.tile(self.fc, (self.Npts, 1))
+
+        self.selector = LassoSelector(ax, onselect=self.onselect)
+
+        self.ind: Optional[NDArrayA] = None
+
+    def export(self, adata: AnnData) -> None:
+        adata.obs["0_LASSO_SELECTED"] = self.exported_data
+
+    def onselect(self, verts: List[NDArrayA]) -> None:
+
+        self.path = Path(verts)
+        self.ind = np.nonzero(self.path.contains_points(self.xys))[0]
+
+        self.fc[:, -1] = self.alpha_other
+        self.fc[self.ind, -1] = 1
```

Comment:
what is this line doing?

Reply:
Line 96 sets the alpha of the unselected coordinates to alpha_other (0.3 is the default). 
Line 97 sets the alpha of selected coordinates to 1 (max possible value) 

I'll include this in the comments ^

Reply:
fantastic yes comment is good